v2/tools/asoctl/internal/crd/cleaner_test.go

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"testing"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/controllers"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake2 "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// TODO: Currently we need to create clientsets for each test as they run in parallel and we run into `resource already exists` error.
+// TODO: We may require a testing suite re-use the clientsets efficiently.
+func getClientSets() (apiextensions.ApiextensionsV1Interface, client.WithWatch, *Cleaner) {
```

Comment:
super duper mega minor (feel free to ignore this if you want): My personal style here is that 3+ argument return functions are a bit wonky. Sometimes they make the most sense but sometimes it's worth defining a little structure that holds the 3 returns.

The reason I like that is:
1. Caller has a field named `fakeAPIExtClient`, rather than needing to know that the 1st argument is that
2. It makes it more clear when checking for errors if the last argument is/isn't an error.

but obviously this is personal preference so no worries if you wanna leave this like it is.

If you had 4 fields and an err (for 5 fields), I'd turn that into a struct with 4 fields and then return `(struct, err)`. In this case because you have no `err` I'd do a struct w/ 3 fields.